eldritch/encounters.py

```diff
@@ -129,22 +129,137 @@ def Lodge1(char):
   check = events.Check(char, "lore", -1)
   draw = events.Draw(char, "spells", 2)
   return events.PassFail(char, check, draw, events.Nothing())
+def Lodge2(char):
+  check = events.Check(char, "fight", -1)
+  ally = events.DrawSpecific(char, "allies", "Thief") # TODO: prereq on the ally being in the deck
+  return events.PassFail(char, check, ally, events.Nothing())
+def Lodge3(char):
+  check = events.Check(char, "luck", -1)
+  curse = events.BlessCurse(char, False)
+  return events.PassFail(char, check, events.Nothing(), curse)
+def Lodge4(char):
+  prereq = events.AttributePrerequisite(char, "dollars", 3, "at least")
+  pay = events.Loss(char, {"dollars": 3})
+  check = events.Check(char, "will", -1)
+  damage = events.Loss(char, {"stamina": 2})
+  membership = events.MembershipChange(char, True)
+  resist = events.PassFail(char, check, events.Nothing(), damage)
+  move = events.ForceMovement(char, "FrenchHill")
+  choice = events.BinaryChoice(char, "Join the Lodge?", "Yes", "No",
+                               events.Sequence([pay, membership], char), resist)
+  return events.Sequence([events.PassFail(char, prereq, choice, resist), move], char)
 def Lodge5(char):
   check = events.Check(char, "lore", -1)
   gain = events.Gain(char, {"clues": 3})
   loss = events.Loss(char, {"clues": float("inf")})
   move = events.ForceMovement(char, "FrenchHill")
   return events.PassFail(char, check, gain, events.Sequence([loss, move], char))
+def Lodge6(char):
+  # Hey, it has the same text!
+  return Lodge4(char)
+def Lodge7(char):
+  check = events.Check(char, "sneak", -2)
+  common1 = events.Draw(char, 'common', 1)
+  common2 = events.Draw(char, 'common', 1)
+  unique1 = events.Draw(char, "unique", 1)
+  unique2 = events.Draw(char, "unique", 1)
+  rolls = events.DiceRoll(char, 2)
+  two_common = events.Sequence([common1, common2], char)
+  two_unique = events.Sequence([unique1, unique2], char)
+  one_each = events.Sequence([common1, unique1], char)
+  cond = events.Conditional(
+    char, rolls, "successes", { 0: two_common, 1: one_each, 2: two_unique, }
+  )
+  return events.PassFail(char, check, events.Sequence([rolls, cond], char), events.Nothing())
 
+def Sanctum1(char):
+  check = events.Check(char, "luck", -2)
+  gain = events.Draw(char, "unique", "all")
+  return events.PassFail(char, check, gain, events.Nothing())
+def Sanctum2(char):
+  check = events.Check(char, "luck", -1)
+  spend = events.Loss(char, {"sanity": 1})
+  success = events.Nothing() # TODO: claim a monster on the board as a trophy
+  nothing = events.Nothing()
+  seq = events.Sequence([
+    spend, events.PassFail(char, check, success, nothing)
+  ])
+  choice = events.BinaryChoice(char, "Cast a banishment spell?", "Yes", "No", seq, nothing)
```

Comment:
Hmm. Does this card actually give you a choice in the matter? It says "spend one sanity" instead of "you may spend one sanity" so it's not clear to me.